### DISCUSSION TITLE
Maven build fails

### DISCUSSION THREAD

**Discussion by hvbtup** [eclipse-birt/birt#736] [Q&A] [answered]
Maven build fails for me with error message:

[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.eclipse.birt.report.designer.all 4.9.0.qualifier
[ERROR]   Missing requirement: org.eclipse.birt.report.designer.all 4.9.0.qualifier requires 'org.eclipse.equinox.p2.iu; org.eclipse.mylyn.bugzilla_feature.feature.group 0.0.0' but it could not be found

### Here's what I was exactly doing

c:\prog\eclinst\birt-master3 was created with Oomph setup.
Then I locally pulled the fix from Claes for issue 649.

```
C:\prog\eclinst\birt-master3\git\birt>pushd git\birt
C:\prog\eclinst\birt-master3\git\birt>git status
On branch Report_viewer_failing_again
nothing to commit, working tree clean

C:\prog\eclinst\birt-master3\git\birt>popd
C:\prog\eclinst\birt-master3>build.cmd

```

### File build.cmd

```
@echo off
setlocal
call mvn-umgebung.cmd
if exist git\birt (
    pushd git\birt
    del /F /S /Q build > NUL:
    rmdir /S /Q build
    if exist build rmdir /S /Q build
    git checkout build
    popd
) else (
    git clone https://github.com/eclipse/birt.git
)

if not exist c:\prog\m2\toolchains.xml copy toolchains.xml c:\prog\m2\
pushd git\birt
rem mvn --global-settings c:\prog\m2\settings.xml --global-toolchains c:\prog\m2\toolchains.xml clean package -DskipTests
mvn --global-settings c:\prog\m2\settings.xml --global-toolchains c:\prog\m2\toolchains.xml build -DskipTests
popd
echo Das fertige Ecclipse sollte in folgendem Verzeichnis zu finden sein:
echo   birt\build\birt-packages\birt-report-all-in-one\targets\products
endlocal

```

### File mvn-umgebung.cmd

```
set JAVA_HOME=c:\prog\jdk\jdk-11.0.13+8
set Path=c:\prog\PortableGit\bin;c:\prog\apache-maven-3.6.1\bin;%JAVA_HOME%\bin;%Path%
set JAVA_TOOL_OPTIONS=-Xmx1024m -Xms1024m
```

I'm using Maven 3.6.1 instead of 3.6.3 - However, I don't think this is related to the issue.

### File toolchains.xml

```
<?xml version="1.0" encoding="UTF8"?>
<toolchains>
  <toolchain>
    <type>jdk</type>
    <provides>
      <version>1.8</version>
	  <id>JavaSE-1.8</id>
    </provides>
    <configuration>
      <jdkHome>c:\prog\jdk\jdk1.8.40\jre</jdkHome>
    </configuration>
  </toolchain>
  <toolchain>
    <type>jdk</type>
    <provides>
      <version>11</version>
	  <id>JavaSE-11</id>
    </provides>
    <configuration>
      <jdkHome>c:\prog\jdk\jdk-11.0.13+8</jdkHome>
    </configuration>
  </toolchain>
</toolchains>
```

Probably this is a silly mis-configuration on my side, but I have no clue.

**Comment by wimjongman** [comment, marked as answer]
The build is run like this:

    mvn package -DskipTests

**Comment by hvbtup** [comment]
Same result. But maybe something is wrong with my setup.
@wimjongman Does the mvn build work for you?

**Reply by wimjongman** [reply]
The pipeline is running fine. Are you compiling the stock BIRT or did you make changes to any of the features (e.g. adding Mylyn)

**Comment by ruspl-afed** [comment]
Maven build works for pipeline, please try to find what is wrong with your configuration comparing to https://github.com/eclipse/birt/blob/master/.github/workflows/ci.yml

**Comment by hvbtup** [comment]
I still don't know what's wrong with my MVN from the command line - and no, I didn't make any changes.

Anyway, I can run MVN from within the Oomph generated Eclipse now, thus I will ignore this issue.

**Reply by wimjongman** [reply]
Did you try the latest maven and removing the toolchains file?

**Comment by hvbtup** [comment]
Let's not waste brain resources with this issue.

**Reply by wimjongman** [reply]
It should work out of the box.

**Reply by SteveSchafer-Innovent** [reply]
It's not working for me as well.  I'm attempting to build a fresh clone.  Maven command:
mvn clean install --no-transfer-progress -U -DskipTests
I also got rid of toolchains by renaming it to toolchains.xml_donotuse.
It's getting a lot of tracebacks because it can't find HttpServletRequest.
It fails on org.eclipse.birt.report.viewer_4.9.0-SNAPSHOT
[build.log](https://github.com/eclipse/birt/files/7670961/build.log)
built with mvn 3.6.3-13 which is the latest available in the Fedora repository, but also tried it with mvn 3.8.4 downloaded from apache with the same result
This worked a couple of months ago.

**Reply by hvbtup** [reply]
"(pre-compile-jsp)" It is the JSP precompilation which fails.
BTW: is it really necessary to pre-compile the JSPs?
IIRC @claesrosell mentioned something like this in another issue.

**Reply by claesrosell** [reply]
I think that it is necessary to pre-compile the jsp pages for some deployment scenarios. But I am not sure. It is not used when the WebViewer is ran from RCP.

@SteveSchafer-Innovent regarding your build error, I got that error years ago. The build was working and suddenly, when changing something unrelated, the jsp pre-compilation stopped working. I never got it to work again, until I build it on another machine.

**Reply by SteveSchafer-Innovent** [reply]
@claesrosell I don't have another machine handy but I was able to build it successfully as a different linux user.  That means it must have been something in my .m2 folder that was messing it up.  I'll investigate and report my findings.

**Reply by SteveSchafer-Innovent** [reply]
Well it wasn't the .m2 folder.  I renamed the .m2 folders in both user home folders so the .m2 folders got re-created. One still succeeded and the other still failed.  The permissions of every folder from root to home folder is the same with respect to the users.  What else could be different?

**Reply by SteveSchafer-Innovent** [reply]
I re-cloned, re-deleted .m2, and this time it built successfully, and now my fork is building too.  Ugh.  It's a mystery.   Well no more brain resources wasted on this one, at least until it shows up again.
